disnake/asset.py

```diff
@@ -168,6 +168,14 @@ async def to_file(
         """
         data = await self.read()
         filename = filename or yarl.URL(self.url).name
+
+        # if the filename doesn't have an extension (e.g. widget member avatars),
+        # try to infer it from the data
+        if not os.path.splitext(filename)[1]:
+            ext = utils._get_extension_for_image(data)
+            if ext:
+                filename += ext
```

Comment:
We should only infer it if the filename wasn't provided as this is a breaking change otherwise. This would change the filename if a user provided `hello` as the filename, for example.